src/xmiddleware.cpp

```diff
@@ -37,4 +41,40 @@ namespace xeus
     {
         return 1000;
     }
+
+    void init_socket(zmq::socket_t& socket,
+                     const std::string& transport,
+                     const std::string& ip,
+                     std::string& port)
```

Comment:
The port parameter should be constant. Port numbers can be retrieved later with `socket.getsockopt( ZMQ_LAST_ENDPOINT, &port, &size );`